shell/core/plugins.js

```diff
@@ -64,12 +64,19 @@ export default function({
     // Load a plugin from a UI package
     loadAsync(id, mainFile) {
       return new Promise((resolve, reject) => {
+        // The plugin is already loaded so we should avoid loading it again.
+        // This will primarily affect plugins that load prior to authentication and we attempt to load again after authentication.
+        if (document.getElementById(id)) {
```

Comment:
This flow needs to be checked with switching between harvester clusters of different versions
- user visits harvester cluster version x, which loads extension version x
- user then visits harvester cluster version y, which i think ends up here with the version x still loaded 

It's been a while, this might work fine as is (can't remember if we unload on exit of harvester product, or specifically call unload when loading a different version.

This change though might resolve the issue of the console warning message we show when on load of the dashboard we load extensions... but try to unload them first... and warn that it doesn't exist

Reply:
I haven't pulled this out of draft yet lol.... I'm glad you commented though, I was going to add a few review comments to have you review around this particular area. I'll go ahead and make the other comments and message you after

Reply:
As far as this particular area, if the Harvester js file name/id includes the version number I expect this to work, if it doesn't then I don't.

> user then visits harvester cluster version y, which i think ends up here with the version x still loaded

Unless Harvester is different in this regard, all of our other extension end up as js file that includes a function which is immediately executed and pollutes the context of the current webpage. Removing the script element doesn't "unexecute" that function and clean up the pollution. 

The only way to make sure the pollution has been cleaned up is to do a full page load which is what we now do when logging the user out. We may have to do the same for harvester if there's a concern that the context will collide with one another.

Reply:
Using the instance @torchiaf provided I was able to verify that multiple versions of harvester still load fine. I couldn't check to see if the extension still loaded on the login page because it appears to be an older version of rancher that doesn't have the new endpoints, but that was verified on a newer version of rancher so I think it'll be fine.


https://github.com/rancher/dashboard/assets/55104481/ff15ebbd-b047-453d-bfb2-90e998f3ac00